test_conformance/geometrics/test_geometrics.cpp

```diff
@@ -265,10 +268,11 @@ int test_geom_cross(cl_device_id deviceID, cl_context context, cl_command_queue
 
     if(!is_extension_available(deviceID, "cl_khr_fp64")) {
         log_info("Extension cl_khr_fp64 not supported; skipping double tests.\n");
-        return 0;
+        return error;
     } else {
         log_info("Testing doubles...\n");
-        return test_geom_cross_double( deviceID,  context,  queue,  num_elements, seed);
+        error |= test_geom_cross_double( deviceID,  context,  queue,  num_elements, seed);
+        return error;
```

Comment:
Have you tried to refactor this in a similar way to what you did for `test_degrees`?

Reply:
will have a look at it